anaconda: Use flattened RPM Packages directory when using USB media

This is a backport of an Anaconda 17 patch.

Signed-off by: Rob Woolley <rob.woolley@windriver.com>

commit 45363385d74d71142fc1eff4f89e8cfbce218ff9
Author: Brian C. Lane <bcl@redhat.com>
Date:   Wed Apr 25 15:37:32 2012 -0700

    Check /run/initramfs/live/ for packages (#806166)
    
    An iso that had been dd'd to a USB stick needs to find the
    packages so look at the root of the media (which dracut mounts
    on /run/initramfs/live/) and bind mount that to /mnt/install/source

Index: b/pyanaconda/yuminstall.py
===================================================================
--- a/pyanaconda/yuminstall.py
+++ b/pyanaconda/yuminstall.py
@@ -483,11 +483,17 @@ class AnacondaYum(yum.YumBase):
                 self._mountInstallCD()
                 self.mediagrabber = self.mediaHandler
                 self._baseRepoURL = "file://%s" % self.tree
+        elif os.path.isdir("/run/initramfs/live/repodata"):
+            # No methodstr was given.  In order to find an installation source
+            # we first check to see if dracut has already mounted the source
+            # on /run/initramfs/live/ and if not we check to see if there's a
+            # CD/DVD with packages on it. If both those fail we default to the
+            # mirrorlist URL.  The user can always change the repo with the
+            # repo editor later.
+            isys.mount("/run/initramfs/live/", self.tree, bindMount=True)
+            self.mediagrabber = self.mediaHandler
+            self._baseRepoURL = "file://%s" % self.tree
         else:
-            # No methodstr was given.  In order to find an installation source,
-            # we should first check to see if there's a CD/DVD with packages
-            # on it, and then default to the mirrorlist URL.  The user can
-            # always change the repo with the repo editor later.
             cdr = scanForMedia(self.tree, self.anaconda.storage)
             if cdr:
                 self.mediagrabber = self.mediaHandler
